url47ebb27197_asirbu_CorrectiveEvolution_tgz-pJ8-eu_fbk_soa_evolution_engine_impl_test_ProblemToSMVFuncTestJ8/src/eu/fbk/soa/evolution/sts/minimizer/TraceEquivalenceMinimizer.java

```diff
@@ -0,0 +1,57 @@
+package eu.fbk.soa.evolution.sts.minimizer;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import org.apache.log4j.Logger;
+
+import eu.fbk.soa.evolution.sts.STS;
+import eu.fbk.soa.util.ConfigUtils;
+
+
+public class TraceEquivalenceMinimizer implements STSMinimizer {
+
+	static Logger logger = Logger.getLogger(TraceEquivalenceMinimizer.class);
+		
+	@Override
+	public void minimizeSTS(STS inputSTS, File outputFile) {
+		String input2Dot = inputSTS.toDot();
+		
+		try {
+			File tempFile = File.createTempFile("temp", "dot");
+//			tempFile = new File("temp.dot");
+			FileWriter fout = new FileWriter(tempFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fout*. The resource is referenced at lines 25, 26. The resource is closed at line 27. There are other execution paths that do not contain closure statements, for example, when OutputStreamWriter.flush throws an exception. Ensure *fout* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).